arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -48,6 +71,94 @@ static void _region_init(u32_t index, u32_t region_base,
 		    SYSMPU->WORD[index][3]);
 }
 
+/* ARM Core MPU Driver API Implementation for NXP MPU */
+
+/**
+ * @brief enable the MPU
+ */
+void arm_core_mpu_enable(void)
+{
+	if (nxp_mpu_enabled == 0) {
+		/* Enable MPU */
+		SYSMPU->CESR |= SYSMPU_CESR_VLD_MASK;
+
+		nxp_mpu_enabled = 1;
+	}
+}
+
+/**
+ * @brief disable the MPU
+ */
+void arm_core_mpu_disable(void)
+{
+	if (nxp_mpu_enabled == 1) {
+		/* Disable MPU */
+		SYSMPU->CESR &= ~SYSMPU_CESR_VLD_MASK;
+		/* Clear Interrupts */
+		SYSMPU->CESR |=  SYSMPU_CESR_SPERR_MASK;
+
+		nxp_mpu_enabled = 0;
+	}
+}
+
+/**
+ * @brief configure the base address and size for an MPU region
+ *
+ * @param   type    MPU region type
+ * @param   base    base address in RAM
+ * @param   size    size of the region
+ */
+void arm_core_mpu_configure(u8_t type, u32_t base, u32_t size)
+{
+	SYS_LOG_DBG("Region info: 0x%x 0x%x", base, size);
+	/*
+	 * The new MPU regions are are allocated per type after the statically
+	 * configured regions.
+	 */
+	u32_t region_index = mpu_config.num_regions + type;
+	u32_t region_attr = _get_region_attr_by_type(type);
+	u32_t last_region = _get_num_regions() - 1;
+
+	/*
+	 * The NXP MPU manages the permissions of the overlapping regions
+	 * doing the logic OR in between them, hence they can't be used
+	 * for stack/stack guard protection. For this reason the last region of
+	 * the MPU will be reserved.
+	 */
+	/* NXP MPU supports up to 16 Regions */
+	if (region_index > _get_num_regions() - 2) {
+		return;
+	}
+
+	/* Configure SRAM_0 region */
+	_region_init(mpu_config.sram_region + 1,
```

Comment:
Because the **mpu_config.sram_region** contains the index of the region in the **mpu_config.mpu_regions** array but the region 0 on the NXP MPU is the background region so on this MPU the regions are mapped starting from 1 hence the **mpu_config.sram_region** on the datastructure is mapped on the **mpu_config.sram_region + 1** region of the MPU.